### DISCUSSION TITLE
Problem with JWT/Builder

### DISCUSSION THREAD

**Discussion by R3B4T0** [lcobucci/jwt#744] [Q&A] [answered]
The error: `Cannot instantiate interface Lcobucci\JWT\Builder`

My program:

```php
use Lcobucci\JWT\Signer\Hmac\Sha256;
use Lcobucci\JWT\Signer\Key;
use Lcobucci\JWT\Builder;

public function chat()
    {
        $username = $this->getUser()->getUsername();
        $token = (new Builder())
            ->withClaim('mercure', ['subscribe' => [sprintf("/%s", $username)]])
            ->getToken(
                new Sha256(),
                new Key($this->getParameter('mercure_secret_key'))
            )
        ;

        $response = $this->render('conversacion/index.html.twig', [
            'controller_name' => 'UsuarioController'
        ]);

        $response->headers->setCookie(
            new Cookie(
                'mercureAuthorization',
                $token,
                (new \DateTime())
                ->add(new \DateInterval('PT2H')),
                '/.well-known/mercure',
                null,
                false,
                true,
                false,
                'strict'
            )
        );

        return $response;
    }
```

**Comment by Ocramius** [comment, marked as answer]
You want to `use Lcobucci\JWT\Token\Builder`, not `use Lcobucci\JWT\Builder`

**Reply by R3B4T0** [reply]
Thank you for your help!

**Reply by lcobucci** [reply]
It looks like you're migrating from v3 to v4, maybe https://lcobucci-jwt.readthedocs.io/en/stable/upgrading/ can help =)

**Reply by mehad022** [reply]
thank you soooo much for referring version migration guide here, **it help me a lot!**

**Reply by lcobucci** [reply]
Glad to hear that! It's quite some work to keep that guide relevant ❤️

**Comment by trobillard** [comment]
I might follow the same auto as R3B4T0 because I have the same code but after changed use Lcobucci\JWT\Token\Builder I get : Too few arguments to function Lcobucci\JWT\Token\Builder::__construct(), 0 passed in /Users/thibaudrobillardDesktop/AFPA/lbchat/src/Controller/IndexController.php on line 20 and exactly 2 expected
I can't find the solution please help

**Reply by lcobucci** [reply]
That's a different problem, probably you have the new major version but didn't update your code.

The builder has required parameters as of v4. The config object makes it easier to instantiate it. However if you don't want to use it you will need to pass the necessary types.

**Reply by trobillard** [reply]
> That's a different problem, probably you have the new major version but didn't update your code.
> 
> The builder has required parameters as of v4. The config object makes it easier to instantiate it. However if you don't want to use it you will need to pass the necessary types. 

Can the problem come from mercure ?

**Reply by lcobucci** [reply]
> /Users/thibaudrobillardDesktop/AFPA/lbchat/src/Controller/IndexController.php on line 20

It seems like it comes from your code

**Reply by pierremartel** [reply]
I have the same problem that @trobillard , i tryed to use the documentation lcobucci but i cannot find a solution,  it's been 1 week i am worked on it...  Can you help me to upgrade my code with lcobucci v4, please @lcobucci

**Comment by programmerShinobi** [comment]
You want to ```use Lcobucci\JWT\Token\Builder```, not ```use Lcobucci\JWT\Builder```

this worked
